tests/integration/targets/hcloud_server_info/tasks/test.yml

```diff
@@ -1,127 +1,89 @@
-- name: verify test gather hcloud server infos with wrong name
-  assert:
+    name: "{{ hcloud_server_name }}-invalid"
+  register: result
+- name: Verify hcloud_server_info with wrong name
+  ansible.builtin.assert:
     that:
-      - server.hcloud_server_info | list | count == 0
+      - result.hcloud_server_info | list | count == 0
 
-- name: test gather hcloud server infos with correct id
+- name: Gather hcloud_server_info with correct label selector
   hetzner.hcloud.hcloud_server_info:
-    id: "{{main_server.hcloud_server.id}}"
-  register: server
-- name: verify test gather hcloud server infos with correct id
-  assert:
+    label_selector: "key=value"
+  register: result
+- name: Verify hcloud_server_info with correct label selector
+  ansible.builtin.assert:
     that:
-      - server.hcloud_server_info|selectattr('name','equalto','{{ hcloud_server_name }}') | list | count == 1
+      - >
+        result.hcloud_server_info
+        | selectattr('name', 'equalto', '{{ hcloud_server_name }}')
+        | list | count == 1
 
-- name: test gather hcloud server infos with wrong id
+- name: Gather hcloud_server_info with wrong label selector
   hetzner.hcloud.hcloud_server_info:
-    id: "4711"
-  register: server
-  ignore_errors: true
-- name: verify test gather hcloud server infos with wrong id
-  assert:
+    label_selector: "key!=value"
+  register: result
+- name: Verify hcloud_server_info with wrong label selector
+  ansible.builtin.assert:
     that:
-      - server is failed
+      - result.hcloud_server_info | list | count == 0
 
-- name: cleanup
-  hetzner.hcloud.hcloud_server:
-    name: "{{ hcloud_server_name }}"
-    state: absent
-
-- name: create server without ips
-  hetzner.hcloud.hcloud_server:
-    name: "{{ hcloud_server_name }}"
-    server_type: cx11
-    image: ubuntu-22.04
-    state: stopped
-    labels:
-      key: value
-    enable_ipv4: false
-    enable_ipv6: false
-  register: main_server
-- name: verify create server
-  assert:
-    that:
-      - main_server is changed
-      - main_server.hcloud_server.name == "{{ hcloud_server_name }}"
-      - main_server.hcloud_server.server_type == "cx11"
-      - main_server.root_password != ""
-- name: test gather hcloud server infos with correct id
+- name: Gather hcloud_server_info (without ip)
   hetzner.hcloud.hcloud_server_info:
-    id: "{{main_server.hcloud_server.id}}"
-  register: server
-- name: verify test gather hcloud server infos with correct id
-  assert:
+    id: "{{ test_server2.hcloud_server.id }}"
+  register: result
+- name: Verify hcloud_server_info (without ip)
+  ansible.builtin.assert:
     that:
-      - server.hcloud_server_info|selectattr('name','equalto','{{ hcloud_server_name }}') | list | count == 1
-- name: cleanup
-  hetzner.hcloud.hcloud_server:
-    name: "{{ hcloud_server_name }}"
-    state: absent
+      - result.hcloud_server_info | list | count == 1
+      - result.hcloud_server_info[0].name == hcloud_server_name + '2'
```

Comment:
What exactly is "Gather hcloud_server_info (without ip)" this testing?

Reply:
The test is part of a bug fix: https://github.com/ansible-collections/hetzner.hcloud/commit/6a7088cf42dcbf01d3c65617db93bff932752cb3